component/exporter.go

```diff
@@ -157,6 +157,13 @@ func WithLogsExporter(createLogsExporter CreateLogsExporterFunc) ExporterFactory
 	}
 }
 
+// WithExporterStabiityLevel overrides the default "unmaintained" stability level.
+func WithExporterStabilityLevel(sl map[config.Type]StabilityLevel) ExporterFactoryOption {
```

Comment:
Can this be just `WithStabilityLevel`? Another idea is to change `WithTracesExporter` to accept a `StabilityLevel` as well?

Reply:
> Can this be just WithStabilityLevel?

The problem with this is that e.g. the receiver factory options also live in this package, so when we want to add this for receivers the name would clash.

Reply:
We can do a trick, the `WithStabilityLevel` can return a type that is all option types Receiver/Exporter/Etc.

Reply:
Also,

> Another idea is to change WithTracesExporter to accept a StabilityLevel as well?

sounds better to me, but I am not sure how we do this without breaking changes

Reply:
2 step, `WithTracesExporterWithStability`?

Reply:
So it sounds like we can go either with:
1. adding the stability level to the traces exporter func
2. adding a WithStabilityLevel and keep WithTracesExporter (and all the other functions like it) as is

@mx-psi @bogdandrutu  I think it makes sense to have this be a parameter that all components will use, which would make option 2 make more sense.

Reply:
The reason I like (1) more than (2) is that cases like https://github.com/open-telemetry/opentelemetry-collector/pull/5580#discussion_r907060692 are detected at compile time (missing/extra argument on a function) instead of at factory build time, but I don't feel strongly about it.

Reply:
@mx-psi @bogdandrutu updated the PR with the suggestion of going w/ `WithTracesExporterAndStabilityLevel`. I think this is the right way to move forward. If folks agree, I will continue w/ the TODO items in the description

Reply:
I agree that this is the right way to move forward 👍

Reply:
Done